packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart

```diff
@@ -330,9 +417,10 @@ class TimelineController
     _httpTimelineLoggingEnabledNotifier.value = state;
   }
 
-  Future<void> clearData() async {
-    if (serviceManager.hasConnection) {
+  Future<void> clearData({bool clearVmTimeline = true}) async {
```

Comment:
a quick comment could be helpful indicating when you should set clearVmTimeline to false.